src/main/java/randoop/reflection/SignatureParser.java

```diff
@@ -0,0 +1,175 @@
+package randoop.reflection;
+
+import java.lang.reflect.AccessibleObject;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/** Parses type signature strings used to identify methods and constructors in input. */
+public class SignatureParser {
+
+  // TODO: The duplicated regular expressions should be factored into a separate source set (aka,
+  // module) so that it can also be used in javagents. The patterns are duplicated from {@code
+  // ReplacementFileReader} from the mapcall agent.
+
+  /** Regex for Java identifiers */
+  private static final String ID_STRING =
+      "\\p{javaJavaIdentifierStart}\\p{javaJavaIdentifierPart}*";
+
+  /**
+   * Regex to match a sequence of identifiers (or {@code <init>}) separated by periods. Corresponds
+   * to package names, fully-qualified classnames, or method names with fully-qualified classname.
+   */
+  private static final String DOT_DELIMITED_IDS =
+      ID_STRING + "(?:\\." + ID_STRING + ")*" + "(?:\\.<init>)?";
+
+  /**
+   * Naive regex to match a method signature consisting of a fully-qualified method name followed by
+   * anything in parentheses. The parentheses are expected to contain argument types, but the
+   * pattern permits anything.
+   *
+   * <p>Capturing group 1 matches the fully-qualified method name, and capturing group 2 matches the
+   * contents of the parentheses.
+   */
+  private static final Pattern SIGNATURE_PATTERN =
+      Pattern.compile("(" + DOT_DELIMITED_IDS + ")\\(([^)]*)\\)");
+
+  /**
+   * Parses a fully-qualified signature and returns the corresponding {@code
+   * java.lang.reflect.AccessibleObject}.
+   *
+   * <p>A signature is expected to have the form
+   *
+   * <ul>
+   *   <li>{@code package-name.classname.method-name(argument-list)} for a method,
+   *   <li>{@code package-name.classname.<init>(argument-list)} or {@code
```

Comment:
It is not that it is discouraged, it is just that for simplicity the documentation only uses the non-`<init>` form.